File: wheystore/wheystore.Web/Modules/SanPhamDB/SanPhamBienThe/SanPhamBienTheForm.cs
namespace wheystore.SanPhamDB.Forms;

[FormScript("SanPhamDB.SanPhamBienThe")]
[BasedOnRow(typeof(SanPhamBienTheRow), CheckNames = true)]
public class SanPhamBienTheForm
{
    public int SanPhamId { get; set; }
    public string TenBienThe { get; set; }
    public decimal GiaBan { get; set; }
    public decimal GiaKhuyenMai { get; set; }
    public int SoLuong { get; set; }
    public bool IsDefault { get; set; }
}

File: wheystore/wheystore.Web/Modules/ServerTypes/SanPhamDB/SanPhamBienTheForm.ts
import { BooleanEditor, DecimalEditor, initFormType, IntegerEditor, PrefixedContext, ServiceLookupEditor, StringEditor } from "@serenity-is/corelib";

export interface SanPhamBienTheForm {
    SanPhamId: ServiceLookupEditor;
    TenBienThe: StringEditor;
    GiaBan: DecimalEditor;
    GiaKhuyenMai: DecimalEditor;
    SoLuong: IntegerEditor;
    IsDefault: BooleanEditor;
}

export class SanPhamBienTheForm extends PrefixedContext {
    static readonly formKey = 'SanPhamDB.SanPhamBienThe';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!SanPhamBienTheForm.init) {
            SanPhamBienTheForm.init = true;

            var w0 = ServiceLookupEditor;
            var w1 = StringEditor;
            var w2 = DecimalEditor;
            var w3 = IntegerEditor;
            var w4 = BooleanEditor;

            initFormType(SanPhamBienTheForm, [
                'SanPhamId', w0,
                'TenBienThe', w1,
                'GiaBan', w2,
                'GiaKhuyenMai', w2,
                'SoLuong', w3,
                'IsDefault', w4
            ]);
        }
    }
}